features/error_reporting.feature

```diff
@@ -1,146 +1,99 @@
-            Notice: Undefined offset: 0 in features/bootstrap/FeatureContext.php line 26
+    001 Scenario: Access undefined index # features/php_errors_in_scenario.feature:9
+          When I access array index 0    # features/php_errors_in_scenario.feature:10
+            Notice: Undefined offset: 0 in features/bootstrap/FeatureContext.php line 24
 
-    2 scenarios (1 passed, 1 failed)
-    7 steps (5 passed, 1 failed, 1 skipped)
+    002 Scenario: Trigger PHP deprecation # features/php_errors_in_scenario.feature:18
+          When I trim NULL                # features/php_errors_in_scenario.feature:19
+            Deprecated: trim(): Passing null to parameter #1 ($string) of type string is deprecated in features/bootstrap/FeatureContext.php line 54
+
+    3 scenarios (1 passed, 2 failed)
+    9 steps (6 passed, 2 failed, 1 skipped)
+    """
+
+  Scenario: With error reporting ignoring E_NOTICE, E_WARNING and E_DEPRECATED
+    When I run behat with the following additional options:
+      | option                                  | value                |
+      | --profile                               | ignore-all-but-error |
+      | features/php_errors_in_scenario.feature |                      |
+    Then it should pass with:
+    """
+    .......
+
+    3 scenarios (3 passed)
+    9 steps (9 passed)
     """
 
-  Scenario: With error reporting ignoring E_NOTICE and E_WARNING
-    Given a file named "behat.yml" with:
-      """
-      default:
-        calls:
-          error_reporting: 32757
-      """
-    When I run "behat -f progress --no-colors features/e_notice_in_scenario.feature"
-    Then it should pass
+  Scenario: With error reporting only ignoring deprecations
+    When I run behat with the following additional options:
+      | option                                  | value   |
+      | --profile                               | ignore-deprecations |
+      | features/php_errors_in_scenario.feature |         |
+    Then it should fail with:
+    """
+    --- Failed steps:
+
+    001 Scenario: Access undefined index # features/php_errors_in_scenario.feature:9
+          When I access array index 0    # features/php_errors_in_scenario.feature:10
+            Notice: Undefined offset: 0 in features/bootstrap/FeatureContext.php line 24
+
+    3 scenarios (2 passed, 1 failed)
+    9 steps (7 passed, 1 failed, 1 skipped)
+    """
 
   Scenario: With very verbose error reporting
-    When I run "behat -f progress --no-colors -vv features/exception_in_scenario.feature"
-    Then it should fail
-    And the output should contain:
+    When I run behat with the following additional options:
+      | option                                 | value   |
+      | --profile                              | default |
+      | -vv                                    |         |
+      | features/exception_in_scenario.feature |         |
+    Then it should fail with:
```

Comment:
This scenario isn't actually covering "error_reporting" in the PHP sense at all - it's to do with how we present Exceptions during the steps at different levels of verbosity. It might be better to move it to a separate feature but I haven't done so yet.